server/services/store/sqlstore/data_retention.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package sqlstore
+
+import (
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	sq "github.com/Masterminds/squirrel"
+	_ "github.com/lib/pq" // postgres driver
+	"github.com/mattermost/focalboard/server/model"
+	_ "github.com/mattn/go-sqlite3" // sqlite driver
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+type RetentionTableDeletionInfo struct {
+	Table         string
+	PrimaryKeys   []string
+	BoardIDColumn string
+}
+
+func (s *SQLStore) runDataRetention(db sq.BaseRunner, globalRetentionDate int64, batchSize int64) (int64, error) {
+	s.logger.Info("Start Boards Data Retention",
+		mlog.String("Global Retention Date", time.Unix(globalRetentionDate/1000, 0).String()),
+		mlog.Int64("Raw Date", globalRetentionDate))
+	deleteTables := []RetentionTableDeletionInfo{
+		{
+			Table:         "blocks",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "board_id",
+		},
+		{
+			Table:         "blocks_history",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "board_id",
+		},
+		{
+			Table:         "boards",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "id",
+		},
+		{
+			Table:         "boards_history",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "id",
+		},
+		{
+			Table:         "board_members",
+			PrimaryKeys:   []string{"board_id"},
+			BoardIDColumn: "board_id",
+		},
+		{
+			Table:         "board_members_history",
+			PrimaryKeys:   []string{"board_id"},
+			BoardIDColumn: "board_id",
+		},
+		{
+			Table:         "sharing",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "id",
+		},
+		{
+			Table:         "category_boards",
+			PrimaryKeys:   []string{"id"},
+			BoardIDColumn: "board_id",
+		},
+	}
+
+	subBuilder := s.getQueryBuilder(db).
+		Select("board_id, MAX(update_at) AS maxDate").
+		From(s.tablePrefix + "blocks").
+		GroupBy("board_id")
+
+	subQuery, _, _ := subBuilder.ToSql()
+
+	builder := s.getQueryBuilder(db).
```

Comment:
Should `team_id == 0` be exempt from data rentention?  Currently we only have templates in there, which is covered in the `where` but filtering out team_id == 0 may be more future proof.

Reply:
I do have a where statement for team_id (line 85). Not sure if I'm missing something?
`Where(sq.NotEq{"team_id": "0"}).`